contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -52,6 +56,17 @@ abstract contract TeleporterUpgradeable is Context, ITeleporterReceiver {
      */
     event TeleporterAddressUnpaused(address indexed teleporterAddress);
 
+    // upgradeableNonReentrant modifier makes sure we cannot reenter between receiver calls.
+    // This modifier should be used for TeleporterUpgradeable contracts so that the execution of a
+    // call from one TeleporterMessenger contract cannot call the ReceiveCrossChainMessage function
+    // of a different TeleporterMessenger contract version that then calls into the same contract.
+    modifier upgradeableNonReentrant() {
```

Comment:
can you clarify why we need a separate reentrancy guard here and not OpenZeppelin's?

Reply:
My understanding is that any non-reentrant implementation applies to every function of that contract, so if we used OZ's reentrant guard here and someone inherits from `TeleporterUpgradeable` and puts the OZ `nonReentrant` on, say, `_receiveTeleporterMessage`, it will always revert because the `_ENTERED` flag will be set when you enter `_receiveTeleporterMessage`.